src/MobileRedirection.php

```diff
@@ -0,0 +1,530 @@
+		 * This may require updating the site's privacy policy or getting user consent for GDPR compliance. Nevertheless,
+		 * since the cookie is not used for tracking this may not be necessary.
+		 *
+		 * @since 1.6
+		 *
+		 * @param bool $should_redirect_via_js Whether JS redirection should be used to take mobile visitors to the AMP version.
+		 */
+		return (bool) apply_filters( 'amp_mobile_redirect_via_js', true ); // @todo Consider `amp_mobile_client_side_redirection`.
+	}
+
+	/**
+	 * Get a list of mobile user agents to use for comparison against the user agent from the current request.
+	 *
+	 * Each entry may either be a simple string needle, or it be a regular expression serialized as a string in the form
+	 * of `/pattern/[i]*`. If a user agent string does not match this pattern, then the string will be used as a simple
+	 * string needle for the haystack.
+	 *
+	 * @return string[] An array of mobile user agent search strings (and regex patterns).
+	 */
+	public function get_mobile_user_agents() {
+		// Default list compiled from the user agents listed in `wp_is_mobile()`.
+		$default_user_agents = [
+			'Mobile',
+			'Android',
+			'Silk/',
+			'Kindle',
+			'BlackBerry',
+			'Opera Mini',
+			'Opera Mobi',
+		];
+
+		/**
+		 * Filters the list of user agents used to determine if the user agent from the current request is a mobile one.
+		 *
+		 * @since 1.6
+		 *
+		 * @param string[] $user_agents List of mobile user agent search strings (and regex patterns).
+		 */
+		return apply_filters( 'amp_mobile_user_agents', $default_user_agents );
+	}
+
+	/**
+	 * Determine if mobile redirection is disabled via query param.
+	 *
+	 * @return bool True if disabled, false otherwise.
+	 */
+	public function is_redirection_disabled_via_query_param() {
+		return isset( $_GET[ self::NO_AMP_QUERY_VAR ] ); // phpcs:ignore WordPress.Security.NonceVerification.Recommended
+	}
+
+	/**
+	 * Determine if mobile redirection is disabled via cookie.
+	 *
+	 * @return bool True if disabled, false otherwise.
+	 */
+	public function is_redirection_disabled_via_cookie() {
+		return isset( $_COOKIE[ self::DISABLED_COOKIE_NAME ] );
+	}
+
+	/**
+	 * Sets a cookie to disable mobile redirection for the current browser session.
+	 *
+	 * @return void
+	 */
+	public function set_cookie_to_disable_redirection() {
+		$value    = '1';                                           // Cookie value.
+		$expires  = 0;                                             // Time till expiry. Setting it to `0` means the cookie will only last for the current browser session.
+		$path     = wp_parse_url( home_url( '/' ), PHP_URL_PATH ); // Path.
+		$secure   = is_ssl();                                      // Whether cookie should be transmitted over a secure HTTPS connection.
+		$httponly = false;                                         // Whether cookie should be made accessible only through the HTTP protocol.
```

Comment:
This could be changed to `true` if we clear the cookie via PHP as mentioned above.

Reply:
Done in Done in 7b46bae117a1c5432c972d4d0fa126f8bc4025af